### PR TITLE
Webwork

### PR DESCRIPTION
**Pull request by mkholjuraev**
Resolves: # RHIF-106
This a separate PR to improve paginated Remediation issues fetching by

1. Moving all cumbersome jobs into a web worker
2. creating grouped API requests with time interval
3. limit and offset are introduced into the API requests to speed up the SQL query.
4. disables remediation button, after it is already clicked and the module is being loading 

Please verify that there is a correct number of API requests to remediate all selected entities and that pagination works as it should.

### CODE REVIEW

**Comment by mkholjuraev** [comment]
@RedHatInsights/team-interact I am ready to do a pair reviewing if this helps ;).

File: src/Utilities/RemediationPairs.js
Comment (bastilian) [inline src/Utilities/RemediationPairs.js:79]: I don't think, you need the `await` or the `Promise.all` here. The Promises pushed into the `request` array are already `await`ed for when looping over the groupChunks and pushed into the array.

Comment (mkholjuraev) [reply src/Utilities/RemediationPairs.js:79]: I thought this will ensure all promises to finish their call. I will verify if they are indeed not necessary and update my PR if needed.

Comment (mkholjuraev) [reply src/Utilities/RemediationPairs.js:79]: @bastilian one more benefit that was in my mind was if I wrap all those API calls promises into a single Promise.all, I would ensure that no remaining promises resolve if any one fails. Considering this, I would keep this. (Feel free to suggest a better way if you have)

Comment (bastilian) [reply src/Utilities/RemediationPairs.js:79]: I'm not 100% certain that is how it would behave. If any of the promises would fail/error and can't be awaited for on line 70, wouldn't they raise and error up there before even reaching the `Promise.all`?

Comment (mkholjuraev) [reply src/Utilities/RemediationPairs.js:79]: hmm, yes it makes sense. I will remove this

Comment (mkholjuraev) [reply src/Utilities/RemediationPairs.js]: Done.

File: src/Utilities/useRemediationDataProvider.test.js
Comment (bastilian) [inline src/Utilities/useRemediationDataProvider.test.js:1]: Maybe we can use the react testing library to add tests for this hook? https://react-hooks-testing-library.com

Comment (bastilian) [reply src/Utilities/useRemediationDataProvider.test.js:1]: There are examples of it being used in compliance to test hooks: https://github.com/RedHatInsights/compliance-frontend/blob/master/src/Utilities/hooks/useTableTools/useActionResolver.test.js

Comment (mkholjuraev) [reply src/Utilities/useRemediationDataProvider.test.js:1]: The issue I faced was more related to mocking the import of a webworker using `import.meta.URL`. There is a library that enables mocking it by configuring babel, but this was causing the webwork not to load into the hook. I am still looking for a way to properly test this.

File: src/Utilities/RemediationPairs.js
Comment (gkarat) [inline src/Utilities/RemediationPairs.js:52]: A nit-pick: this seems to be inefficient, especially when you search for indices with `findIndex`. Probably, you could use `reduce` here together with `pairs` set as an object, not array? This will let you easier work and update the dictionary (object). In the end, you can transform it back to array...

**Review by gkarat — APPROVED** [review]
lgtm! tested with the steps you provided, works ok for accounts with lesser number of advisories/systems. a few comments on the code: I think there are places where you can work with data more efficiently by reducing loops

File: src/Utilities/RemediationPairs.js
Comment (mkholjuraev) [reply src/Utilities/RemediationPairs.js:52]: Actually, I started pairs as an object initially but later had to convert it to array type. Let me remember why I switched to Array.

**Pull request merged**
